src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,119 @@
+package duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * Handles storage operations including loading tasks from a file and saving tasks to a file.
+ * <p>
+ * This class provides methods to interact with the storage file where tasks are saved.
+ * It ensures that the necessary directory and file exist and provides utility functions
+ * to read and write tasks from and to the storage file.
+ * </p>
+ */
+public class Storage {
+    private static final String FILE_PATH = "./data/duke.txt";
+
+    /**
+     * Ensures the storage directory exists.
+     * <p>
+     * If the directory does not exist, it is created.
+     * </p>
+     */
+
+    private void ensureDirectoryExists() {
```

Comment:
I like how you name your methods, which is consistent throughout across the classes you have created. It clearly explained the named entity accurately and at a sufficient level of detail.